lib/fluent/plugin/out_splunkhec.rb

```diff
@@ -72,7 +79,11 @@ def write(chunk)
           when Fixnum
             event = record.to_s
           when Hash
-            event = record.to_json.gsub("\"", %q(\\\"))
+            if @event_send_as_json
+              event = record.to_json
+            else
+              event = record.to_json.gsub("\"", %q(\\\"))
+            end
```

Comment:
With the current implementation the Json appears in Splunk as a quoted string. 

With my change, the quoting is skipping and the Json appears in Splunk as a true Json object.